python/ray/data/dataset.py

```diff
@@ -2513,7 +2513,10 @@ def write_json(
         ray_remote_args: Dict[str, Any] = None,
         **pandas_json_args,
     ) -> None:
-        """Writes the :class:`~ray.data.Dataset` to JSON files.
+        """Writes the :class:`~ray.data.Dataset` to JSON and JSONL files.
+
+        When the dataset has multiple rows, the output file is in JSONL format.
+        Otherwise, the output file is in JSON format.
```

Comment:
Nit: IMO it's better to exclude this information, as well the example of writing to a single JSON file. How often will users work with single-row datasets? I'm concerned this would introduce confusion, and it'd be easier to understand "write_json always writes JSONL".